purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -353,29 +353,29 @@ class Purchases internal constructor(
     }
 
     /**
-     * Gets the StoreProduct for the given list of subscription skus.
-     * @param [skus] List of skus
+     * Gets the StoreProduct for the given list of subscription and non-subscription productIds.
+     * @param [productIds] List of productIds
      * @param [callback] Response callback
      */
-    // TODO deprecate, replaced with getProducts
-    fun getSubscriptionSkus(
-        skus: List<String>,
-        callback: GetStoreProductsCallback
-    ) {
-        getSkus(skus.toSet(), ProductType.SUBS, callback)
-    }
-
-    /**
-     * Gets the SKUDetails for the given list of non-subscription skus.
-     * @param [skus] List of skus
-     * @param [callback] Response callback
-     */
-    // TODO deprecate, replaced with getProducts
-    fun getNonSubscriptionSkus(
-        skus: List<String>,
+    fun getProducts(
+        productIds: List<String>,
         callback: GetStoreProductsCallback
     ) {
-        getSkus(skus.toSet(), ProductType.INAPP, callback)
+        getProducts(productIds.toSet(), ProductType.SUBS, object : GetStoreProductsCallback {
```

Comment:
we could avoid the explicit type here and just pass lambdas :)  

```
        getProducts(
            productIds.toSet(), 
            ProductType.SUBS, 
            { subStoreProducts ->
                getProducts(
                    productIds.toSet(), 
                    ProductType.INAPP, 
                    { inAppStoreProducts -> callback.onReceived(subStoreProducts + inappStoreProducts) },
                    { callback.onError(it) }
            }, { callback.onError(it) }
        )
``` 

but with better spacing..

Reply:
This is what I tried doing but couldn't get to work 😩 I tried copying and pasting what you had here and it still wasn't it happy. Was getting an error about it not being able to infer lambda or something? 🤔